pkgs/os-specific/linux/zfs/default.nix

```diff
@@ -47,6 +47,11 @@ let
 
       postPatch = optionalString buildKernel ''
         patchShebangs scripts
+
+        # https://github.com/openzfs/zfs/issues/10107
+        substituteInPlace ./config/kernel.m4 \
+          --replace "make modules" "make CC=$CC modules"
```

Comment:
Can you also try to make this patch upstream?

Reply:
Done: https://github.com/openzfs/zfs/pull/12525